akka-typed/src/main/scala/akka/typed/scaladsl/Actor.scala

```diff
@@ -188,6 +188,20 @@ object Actor {
   }
 
   /**
+   * Construct an actor behavior that can react to lifecycle signals only. After spawning this actor
+   * from another actor (or as the guardian of an [[akka.typed.ActorSystem]]) it will be executed
+   * within an [[ActorContext]] that allows access to the system, spawning and watching other
+   * actors, etc.
+   */
+  def signal[T](
+    init:     ActorContext[T] ⇒ Unit,
+    onSignal: PartialFunction[(ActorContext[T], Signal), Behavior[T]]): Behavior[T] =
+    Actor.deferred { context ⇒
+      init(context)
+      immutable[T]((_, _) ⇒ Actor.empty).onSignal(onSignal)
```

Comment:
I think I agree with @jrudolph: too much magic, unclear to read.

But why `same` instead of `empty`?

Reply:
Good point, @jrudolph . And when thinking some more about it there is another use case than with `deferred`. You can have an actor that starts out as ordinary `immutable` and then changes to only handle some signals, such as waiting for termination.

Would `handleSignals` be `Actor.handleSignals`? Or did you have something else in mind?
Naming wise I think it could still be `Actor.signal`.